ceph_deploy/osd.py

```diff
@@ -372,8 +372,8 @@ def disk_list(args, cfg):
             command,
         )
         for line in out:
-            if line.startswith('Disk /'):
-                distro.conn.logger(line)
+            if line.startswith(b'Disk /'):
```

Comment:
this is somewhat correct, I think the better approach would be to check if the incoming string is a bytearray or a plain string. A utility that can be re-used would be better, as this is not the only place in ceph-deploy where this check would be needed. Something like:

    if isinstance(string, bytes):
        # we really ignore here if we can't properly decode with utf-8
        return string.decode('utf-8', 'ignore')
    return string

Reply:
*maybe* even better would be to make remoto do this automatically for us, optionally turning it off (but decoding to string by default)

Reply:
That would be great, and the actual right way of doing it. I didn't want to send a whole refactor patch :P
Given that `remoto` is your lib, you should decide whether it's gonna be done by it or by this in an aux function. How would you like to proceed?

Reply:
Lets get this done on the remoto side, I'll take care of the packaging. Would you like to try a PR against remoto?

Reply:
I had a heavy week, and will have a heavy one, sorry for the delay.
But yeah, lets get on it. I need to catch up on the code, so probably it's gonna be for the end of the week or so.